src/lib/api.mli

```diff
@@ -138,8 +139,9 @@ module type BASIC_STRING = sig
   (** [make size char] builds a new string of the passed [length] where the
       character at every position is [char], like [String.make].
 
-      @raise Invalid_argument if size is [< 0] or
-      [> {max_string_length}] if it is bounded.*)
+      [make size] may raise an exception when [size] is [< 0] or
+      [> {max_string_length}] (if it is [Some _]) depending on the
+      backend implementing the API. *)
```

Comment:
Also we don't know which exception may be raised in `api.mli`.

I expect `@raise Exn <condition>` where the condition is a definitive logical proposition that tells me when the function may raise the exception.
Here we don't know whether we may raise an exception.